Dockerfile

```diff
@@ -1,28 +1,26 @@
+# Stage 1: Build Go Dependencies
 FROM golang:1.21-alpine3.17 as goDependencies
 
-COPY ./src/go.mod /deps/go.mod
-COPY ./src/go.sum /deps/go.sum
-WORKDIR /deps
+COPY ./src/go.mod /app/go.mod
+COPY ./src/go.sum /app/go.sum
+WORKDIR /app
 RUN go mod download
 
-FROM ghcr.io/foundry-rs/foundry:latest as contracts
-COPY gnosh-contracts /deps/gnosh-contracts
-COPY .git /deps/.git
-COPY src/Makefile /deps/src/Makefile
-WORKDIR /deps/src
-RUN apk add --no-cache make
-RUN make compile-contracts
-
+# Stage 2: Build the Go Application
 FROM golang:1.21-alpine3.17 as appBuilder
```

Comment:
Is there an advantage to splitting the deps into their own stage? Just doing the deps steps in the builder before the actual build should be cached just the same